### DISCUSSION TITLE
Trying to play with plots

### DISCUSSION THREAD

**Discussion by linwryn** [pybamm-team/PyBaMM#2117] [Q&A] [answered]
Hi,
I'm trying to figure out how to put negative electrode potential, positive electrode potential and electrolyte potential in a one figure. Also, I'm curious that how can I change the domain of each parameter? Not like [0,3600] but like changing the μm part. If that domain change is not logical can you explain why?

Thanks.

**Comment by rtimms** [comment, marked as answer]
If two variables have the same domain you can plot them on the same axes by nesting lists, e.g. 
```python3
sim.plot([["Negative electrode current density [A.m-2]", "Negative electrolyte current density [A.m-2]"]])
```

You can pass `spatial_unit` to the plot to change the units, e.g.
```python3
sim.plot([["Negative electrode current density [A.m-2]", "Negative electrolyte current density [A.m-2]"]], spatial_unit="mm")
```

If you want to create custom plots you can always access the underlying arrays by accessing the solution as a dict and doing `.entries`, e.g. `sim.solution["Negative electrode potential [V]"].entries`. This will return an array that you can use to create your own plots however you like.

**Reply by linwryn** [reply]
I want to ask a follow up question for that. So, when I tried to get the data as you suggest I get 1980 data points for both negative electrode potential and positive electrode potential but for electrolyte potential I get 5940. What are these numbers? I first thought that they have to be the data points for each second but apparently they are not. If these were covered in one of the videos on youtube, sorry about that but can you explain what are those numbers? 

Thanks

**Reply by rtimms** [reply]
The size of the array depends on the domain of the variable. For example, if you solved at `N_t` time points and had `N_n` points in the negative electrode, `N_s` points in the separator and `N_p` points in the positive electrode then `sim.solution["Negative electrode potential [V]"].entries` would return an `N_n` times `N_t` size array, and `sim.solution["Electrolyte potential [V]"].entries` would return an `N_n+N_s+N_p` times `N_t` size array.
